### PR TITLE
Adds middle-click close to editor tabs

### PR DESCRIPTION
**Pull request by lukaszsobek**
Associated Issue: #5323

Adds middle click capability to close edit tab.

### CODE REVIEW

File: src/components/Editor/Tab.js [hunk 1]
```diff
@@ -156,11 +156,31 @@ class Tab extends PureComponent<Props> {
...
     const sourceAnnotation = getSourceAnnotation(source, sourceMetaData);
 
-    function onClickClose(ev) {
-      ev.stopPropagation();
+    function onClickClose(e) {
+      e.stopPropagation();
       closeTab(source.get("url"));
     }
 
+    function handleTabClick(e) {
```
Comment (darkwing) [inline src/components/Editor/Tab.js]: I think we can shorten this function up quite a bit:

```js
e.preventDefault();
e.stopPropagation();

if(e.button === 1) {
    return closeTab(source.get("url"));
}

return selectSource(sourceId);
```

**Comment by darkwing** [comment]
Awesome start @lukaszsobek !

File: src/components/Editor/Tab.js (hunk 1 above)
Comment (lukaszsobek) [reply src/components/Editor/Tab.js]: Yeah, definitely.

File: src/components/Editor/Tab.js [hunk 2]
```diff
@@ -156,11 +156,25 @@ class Tab extends PureComponent<Props> {
...
+      e.preventDefault();
+      e.stopPropagation();
+
+      if (e.button === 1) {
+        return closeTab(source.get("url"));
+      }
+
+      return selectSource(sourceId);
+
+      // onMouseUp doesn't overwrite default onContextMenu behavior,
```
Comment (darkwing) [inline src/components/Editor/Tab.js]: Let's either remove this comment or move it to the top of the function.

**Comment by darkwing** [comment]
Updated comments, will merge once Travis passes.

**Comment by darkwing** [comment]
👍

**Pull request merged**